flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredOperatorBackendStateMetaInfo.java

```diff
@@ -175,14 +175,21 @@ public boolean equals(Object obj) {
 				return false;
 			}
 
+			Snapshot snapshot;
+
+			if (obj instanceof Snapshot) {
+				snapshot = (Snapshot)obj;
+			} else {
```

Comment:
nit: if you place the condition for else first, declaration and assignment to snapshot can be done in one line.